python/src/aac/validate/_validate.py

```diff
@@ -0,0 +1,100 @@
+from contextlib import contextmanager
+from iteration_utilities import flatten
+
+from aac.lang.language_context import LanguageContext
+from aac.lang.active_context_lifecycle_manager import get_active_context
+from aac.lang.definitions.definition import Definition
+from aac.lang.definitions.structure import get_sub_definitions
+from aac.lang.hierarchy import get_definition_ancestry
+from aac.parser import parse
+from aac.plugins.plugin_manager import get_validator_plugins
+from aac.plugins.validators import ValidatorPlugin, ValidatorResult
+from aac.validate._validation_error import ValidationError
+from aac.validate._validation_result import ValidationResult
+from aac.validate._collect_validators import get_applicable_validators_for_definition
+
+
+@contextmanager
+def validate_definitions(user_definitions: list[Definition]) -> ValidationResult:
```

Comment:
It looks like another comment that disappeared: Could we rename these context managers to something that reads better after `with` - i.e. `with validate_definitions` doesn't sound right, whereas `with validated_definitions` or something would sound more natural.

Reply:
Done